model/src/channel/message/mod.rs

```diff
@@ -25,41 +25,48 @@ use std::collections::HashMap;
 
 #[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
 pub struct Message {
+    #[serde(skip_serializing_if = "Option::is_none")]
     pub activity: Option<MessageActivity>,
+    #[serde(skip_serializing_if = "Option::is_none")]
     pub application: Option<MessageApplication>,
     pub attachments: Vec<Attachment>,
     pub author: User,
     pub channel_id: ChannelId,
     pub content: String,
     pub edited_timestamp: Option<String>,
     pub embeds: Vec<Embed>,
+    #[serde(skip_serializing_if = "Option::is_none")]
     pub flags: Option<MessageFlags>,
+    #[serde(skip_serializing_if = "Option::is_none")]
     pub guild_id: Option<GuildId>,
     pub id: MessageId,
     #[serde(rename = "type")]
     pub kind: MessageType,
+    #[serde(skip_serializing_if = "Option::is_none")]
     pub member: Option<PartialMember>,
-    #[serde(default)]
+    #[serde(default, skip_serializing_if = "Vec::is_empty")]
```

Comment:
This is just a side-note but it's weird that this is the only mention field that's optional in any way, just leaving this response here to document that I've confirmed it from the docs and an API response in case anyone else questions it.